modules/oxide-components/src/main/ts/internal/icon/Icon.component.tsx

```
@@ -0,0 +1,18 @@
+  /** The function to resolve the icon name to an html string.
+   * This would eventually default to retrieving the icon from the editor's registry.
+   * (name: string) => editor.ui.registry.getAll().icons[name] ?? 'temporary-placeholder'
+   *
+   * @param icon - The name of the icon
+   * @returns The html string representation of the icon
+   */
+  resolver: (icon: string) => string;
+}
+
+export const Icon: React.FC<IconProps> = ({ icon, resolver }) => {
+  return <span className={classes([ 'tox-icon' ])} dangerouslySetInnerHTML={{ __html: resolver(icon) }} />;
```

Biome (2.1.2)

[error] Avoid passing content using the dangerouslySetInnerHTML prop.

Setting content using code can expose users to cross-site scripting (XSS) attacks

(lint/security/noDangerouslySetInnerHtml)